src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from 'react';
+import users from '../../api/users';
+import { ToDos } from '../../react-app-env';
+import TodoInfo from '../TodoInfo/TodoInfo';
+import UserInfo from '../UserInfo/UserInfo';
+import './TodoList.css';
+
+interface Props {
+  initialTodos: ToDos[];
+}
+
+const TodoList: React.FC<Props> = ({ initialTodos }) => {
+  const [todos, setTodos] = useState(initialTodos);
+  const [newTitle, setNewTitle] = useState('');
+  const [selectedUser, setSelectedUSer] = useState(0);
+  const [showTitleError, setShowTitleError] = useState(false);
+  const [showSelectErrorMessage, setShowSelectErrorMessage] = useState(false);
+
+  function addTodo() {
+    const newTodo = {
+      title: newTitle,
+      userId: selectedUser,
+      completed: false,
+      id: todos.length + 1,
+      user: users.find(user => user.id === selectedUser) || null,
+    };
+
+    setTodos([...todos, newTodo]);
+
+    setNewTitle('');
+    setSelectedUSer(0);
+  }
+
+  return (
+    <>
+      <form onSubmit={(event) => {
+        event.preventDefault();
+
+        if (newTitle && selectedUser) {
+          addTodo();
+        }
+      }}
+      >
+        <label htmlFor="titleInput">
+          Title:
+
+          <input
+            id="titleInput"
+            type="text"
+            data-cy="titleInput"
+            placeholder="Title"
+            value={newTitle}
+            onChange={(event) => {
+              setShowTitleError(false);
+
+              const title = event
+                .target
+                .value
+                .split('').filter(character => (
+                  /[a-z0-9]/i.test(character) || /\s/.test(character)
+                ))
+                .join('');
+
+              setNewTitle(title);
+            }}
```

Comment:
let's move it outside JSX - before `return` and just pass it here as a one-line prop
```suggestion
            onChange={handleInputChange}
```